frame/merkle-mountain-range/src/lib.rs

```diff
@@ -196,13 +200,23 @@ pub mod pallet {
 	#[pallet::getter(fn mmr_leaves)]
 	pub type NumberOfLeaves<T, I = ()> = StorageValue<_, LeafIndex, ValueQuery>;
 
-	/// Hashes of the nodes in the MMR.
+	/// New branch added by current block - the leaf (data) and parent nodes (hashes).
+	///
+	/// This is single value storage overwritten by each block. It is used as temporary
+	/// leaf storage until after block is built and offchain worker can move/copy this
+	/// leaf and its parent nodes in offchain database under `node_offchain_key()`.
+	#[pallet::storage]
+	#[pallet::getter(fn new_mmr_branch)]
+	pub type NewBranch<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, MmrBranch<<T as Config<I>>::Hash, LeafOf<T, I>>, OptionQuery>;
+
+	/// Hashes of the peaks in the MMR.
 	///
 	/// Note this collection only contains MMR peaks, the inner nodes (and leaves)
 	/// are pruned and only stored in the Offchain DB.
 	#[pallet::storage]
 	#[pallet::getter(fn mmr_peak)]
-	pub type Nodes<T: Config<I>, I: 'static = ()> =
+	pub type Peaks<T: Config<I>, I: 'static = ()> =
```

Comment:
Do we need to perform runtime upgrade on chains that are using `pallet-mmr` (Rococo at least)? IIUC this change will leave abandoned `Nodes` entries in the runtime storage and `Peaks` will be empty initially.

Reply:
Good point, I've reverted the rename, not worth the effort.